scripts/benchmark.cr

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env -S crystal run --no-debug --release
+require "../src/pluto"
+
+record BenchmarkResult, time : Int64, memory : Int64
+
+def benchmark_memory
+  bytes_before_measure = GC.stats.total_bytes
+  yield
+  (GC.stats.total_bytes - bytes_before_measure).to_i64
+end
+
+def benchmark_time
+  time = Time.monotonic
+  yield
+  Time.monotonic - time
+end
+
+def print_result_table(total : Hash(String, BenchmarkResult))
+  name_rjust = total.keys.map(&.size).max
+  time_ljust = total.values.map { |t| t.time.to_s.size + 2 }.max
+  memo_ljust = total.values.map(&.memory.humanize_bytes.size).max
+
+  # Headers
```

Comment:
What would `headers` contain?

Reply:
Left this as it was since defining a variable that's only used for appending to the `table` variable seemed unnecessary :/

Reply:
I was thinking about something like this:

```crystal
headers = [
  [" " * name_rjust, "Time".ljust(time_ljust), "Memory".ljust(memo_ljust)],
  ["-" * name_rjust, "-" * time_ljust, "-" * memo_ljust],
]
table << headers
```

I agree that an extra variable is not needed, but it makes code self-explanatory (so no need for a comment).

Reply:
`rows` or `table` is currently an `Array(Array(String))`, so it would have to be: 
```
...
table << headers[0]
table << headers[1]
```
Is that acceptable? While I very much like self-explanatory code, it seems simpler to keep it as it is with the single word comment. I'll defer to you though if the above looks good.

(I've kept the variable name `table` so far since tables are already collections of `row`s, and the method is generating a table, but I can change this name too if you think it makes more sense)